javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -109,10 +109,14 @@ private static void ApplyParameterTransformations(JavaBlock function, ClientMeth
                         return parameterName + " != null";
                     }).collect(Collectors.joining(" || "));
             boolean conditionalAssignment = nullCheck != null && !nullCheck.isEmpty() && !transformation.getOutParameter().getIsRequired() && !clientMethod.getOnlyRequiredParameters();
+            // Use a mutable internal variable, leave the original name for effectively final variable
```

Comment:
Use a `fooInternal` to do conditional assignment, leave `foo` as effectively final, for lambda.

Hope it will not get into future naming conflict.